java_gen/templates/of_class.java

```diff
@@ -419,6 +419,42 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    //:: for prop in msg.data_members:
+    //:: if prop.name == 'xid':
+    public boolean equalsIgnoreXid(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        //:: if len(msg.data_members) > 0:
+        ${msg.name} other = (${msg.name}) obj;
+        //:: #endif
+
+        //:: for prop in msg.data_members:
+        //:: if prop.java_type.is_primitive and prop.name == 'xid':
+        // ignore XID
+        //:: elif prop.java_type.is_primitive:
+        if( ${prop.name} != other.${prop.name})
+            return false;
+        //:: elif prop.java_type.is_array:
+        if (!Arrays.equals(${prop.name}, other.${prop.name}))
+                return false;
+        //:: else:
+        if (${prop.name} == null) {
+            if (other.${prop.name} != null)
+                return false;
+        } else if (!${prop.name}.equals(other.${prop.name}))
+            return false;
+        //:: #endif
+        //:: #endfor
+        return true;
+    }
+    //:: break
+    //:: #endif
+    //:: #endfor
+
```

Comment:
Sure; it's really a small nit:

When you look at the artifact delta created from this pull request (the latest commit in the artifact repo that bsn-abat links to, https://github.com/floodlight/loxigen-artifacts/commit/2e58a1ba4849e79e104158d4db47b4651f0666d0), you'll see that *all* classes gain an additional newline (not just the ones that actually have an xid field). The reason is that the newline in L.457 (it has moved up a couple of lines with your latest commit) gets inserted unconditionally.

TL;DR: if you move the newline into the #if/#endif block it should be correct.